src/main.rs

```diff
@@ -14,6 +14,15 @@ mod shell;
 use shell::Shell;
 use shell::ShellKind;
 
+/// Environment variable that indicates that the Nix profile has already been sourced.
```

Comment:
perhaps explicitly mention here that the reason that we're exporting this environment variable is to prevent those profile scripts from being sourced again inside the nix shell and clobbering its `PATH`